### ISSUE TITLE
Version 118 not compatible AOS < 6 camera

### ISSUE THREAD

**Issue by lukeswitz** [guardianproject/haven#355]
Will not show the camera config/monitor views. The migration to androidx libs have issues for devices on lower APIs. 

- Confirmed 117 working on LG Nexus 4 & 5/stock kernels. 

- XML errors from changes likely, shows OOM. Other reason could be resizing the Canvas with a bad value; breaks under the androidx schema. LowMemoryKiller terminates the app on device/simulator.

- Looking closer, the quote is missing on initial PPIntro slide. Calling setText with a bad value would cause this crash. Logs to follow

**Comment by n8fr8** [comment]
Thanks for the legacy device testing. Ideally we can balance moving forward to modern, easier to manage code and modules while ensuring we support the "cheap old second device" still. I had hoped to push out 118 more widely, but alas, seems like we need to fix this pronto.

**Comment by lukeswitz** [comment]
```
10-23 16:09:28.594 3030-3030/org.havenapp.main I/art: Rejecting re-init on previously-failed class java.lang.Class<androidx.core.view.ViewCompat$OnUnhandledKeyEventListenerWrapper>
10-23 16:09:28.594 3030-3030/org.havenapp.main I/art: Rejecting re-init on previously-failed class java.lang.Class<androidx.core.view.ViewCompat$OnUnhandledKeyEventListenerWrapper>
10-23 16:09:28.614 3030-3030/org.havenapp.main W/linker: librsjni_androidx.so: unused DT entry: type 0x6000000f arg 0xaf80
10-23 16:09:28.614 3030-3030/org.havenapp.main W/linker: librsjni_androidx.so: unused DT entry: type 0x60000010 arg 0x1d
10-23 16:09:28.614 3030-3030/org.havenapp.main W/linker: librsjni_androidx.so: unused DT entry: type 0x6ffffef5 arg 0xad5c
10-23 16:09:28.614 3030-3030/org.havenapp.main W/linker: librsjni_androidx.so: unused DT entry: type 0x6ffffffe arg 0xad1c
10-23 16:09:28.614 3030-3030/org.havenapp.main W/linker: librsjni_androidx.so: unused DT entry: type 0x6fffffff arg 0x2
10-23 16:09:28.614 3030-3030/org.havenapp.main A/libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0x0 in tid 3030 (g.havenapp.main)

```

**Comment by lukeswitz** [comment]
It was my assumption this app was to be used on “burners” or a “cheap old second device”.  I have a legacy branch that works aside from video, which never worked and broke RC 1. All in all I’m at a loss as to what this project needs or wants. Simple security was almost there.

Adding a bunch of code libraries that are most definitely not meant for burner devices has halted development. CameraPlus library worked until updated, why it was added instead of writing a little code to support both CameraAPIs?

I’d like to help protect people, yet this app is only providing a false sense of security. It was terrible reading last year the comment from somebody who needed this to work and it didn’t capture any images of the people robbing him. 

The camera is almost all you need working in a “security camera” app. The smart move would be removing all unnecessary libs (dozens working against one another is doing no good)

As it’s past the three month mark without any update from Freedom of the Press, Nate F or otherwise- I have to ask: has the funding for this project ($33k last I looked) been reallocated? Is Haven on the shelf?

Appreciate your response & efforts,

**Comment by n8fr8** [comment]
Hey, luke, and thanks for the continued support and interest. I've been occupied with some other major, but related projects (infrastructure work for Tor, Onion Services, secure messaging, etc), for the last few months. I and other Guardian Project team members have also been doing a lot of work with activists in need of physical protection out in the field. We've been gathering feedback, usability input and more, as part of our broad work, but also on Haven. So, while I haven't been active here on the code, I have been actively observing and reflecting on use of Haven, and the way forward.

Regardless, that isn't an excuse for not ensuring the current app is stable and available for all the platforms we claim to support. I have been working with some new CameraView projects with other apps, and think what we need here is an overall of this particular piece of code. If there is a short term fix for the AOS < 6 issue, I'll get that done first, but in the long run, in order to support video well, and support a broad range of OSes, we need some more robust code in the Camera department.

**Comment by lukeswitz** [comment]
Thanks for all you do (especially out in the field). Exciting to see the repo active again!

**Comment by n8fr8** [comment]
Found and fixed an issue in #361 related to RenderScript through AndroidX being broken on devices below Android SDK 23. It turns out that using RenderScript doesn't buy us much performance improvement, so removing the dependency for now!
